README.md

```diff
@@ -1,12 +1,60 @@
+
+### HTTP
+The C-SDK provides the ability to establish an HTTP connection with a server over a customer-implemented transport layer, which can either be a secure channel like a TLS session (mutually authenticated or server-only authentication) or a non-secure channel like a plaintext TCP connection. The HTTP connection can be used to make REST API requests that include "GET" (include range requests), "PUT", "POST" and "HEAD" requests. The SDK provides a mechanism to register a customer-defined callback for receiving parsed header fields in an HTTP response. The [coreHTTP](https://github.com/FreeRTOS/coreHTTP) library has been refactored for memory optimization, and is a client implementation of a subset of the [HTTP/1.1](https://tools.ietf.org/html/rfc2616) standard.
+
+### AWS IoT Device Shadow
+[AWS IoT Device Shadow](https://github.com/aws/device-shadow-for-aws-iot-embedded-sdk) enables you to store and retrieve the current state (the “shadow”) of every registered device. The device’s shadow is a persistent, virtual representation of your device that you can interact with from AWS IoT Core even if the device is offline. The device state captured as its “shadow” is itself a [JSON](https://www.json.org/) document. The device can send commands over MQTT or HTTP to update its latest state. Each device’s shadow is uniquely identified by the name of the corresponding “thing”, a representation of a specific device or logical entity on the AWS Cloud. See [Managing Devices with AWS IoT](https://docs.aws.amazon.com/iot/latest/developerguide/iot-thing-management.html) for more information. More details about shadows can be found in [AWS IoT documentation](https://docs.aws.amazon.com/iot/latest/developerguide/iot-device-shadows.html). 
+
+### JSON Parser
+The coreJSON library is a JSON parser that strictly enforces the [ECMA-404 JSON standard](https://www.json.org/json-en.html).
+
+
+## Metrics
+
+Within the MQTT Demo, users have the ability to report operating System, hardware platform and MQTT Client information to AWS IoT by sending a specially formatted string in the username field of the MQTT CONNECT packet.  
```

Comment:
This isn't a metric that the customer can consume, is it?

Reply:
What do you mean by "consume"? It is in the demo only and not part of the MQTT library. We found that most customers have no problem adding metrics to their MQTT message to AWS IoT themselves, if they knew how. We let customers know that these metrics help AWS IoT improve security and provide better technical support (this statement is taken from PM justification doc). 


I think we Should we add the text: "These metrics help AWS IoT improve security and provide better technical support."

Reply:
I see a title like "Metric" and I think I will be able to see data about the workings of my system.  In this case the data is for AWS, and not for the customer.  I think we should call it something other than Metric.

Reply:
That's a valid concern. Have updated the section to "Collection of Metrics" and updated the text to explicitly mention that these metrics are for AWS utility, along with providing information on how these can be disabled.

Reply:
Maybe "Collection of Metrics for AWS IoT"? lol